crates/algokit_utils/src/transactions/creator.rs

```diff
@@ -0,0 +1,250 @@
+
+use super::{
+    AppCallMethodCallParams, AppCallParams, AppCreateMethodCallParams, AppCreateParams,
+    AppDeleteMethodCallParams, AppDeleteParams, AppUpdateMethodCallParams, AppUpdateParams,
+    AssetClawbackParams, AssetCreateParams, AssetDestroyParams, AssetFreezeParams,
+    AssetOptInParams, AssetOptOutParams, AssetReconfigureParams, AssetTransferParams,
+    AssetUnfreezeParams, NonParticipationKeyRegistrationParams, OfflineKeyRegistrationParams,
+    OnlineKeyRegistrationParams, PaymentParams,
+    common::TransactionSigner,
+    composer::{Composer, ComposerError},
+};
+
+#[derive(Debug, Clone)]
+pub struct BuiltTransactions {
+    pub transactions: Vec<Transaction>,
+    pub method_calls: HashMap<usize, ABIMethod>,
+    #[debug(skip)]
+    pub signers: HashMap<usize, Arc<dyn TransactionSigner>>,
+}
+
+/// Creates individual Algorand transactions.
+pub struct TransactionCreator {
+    new_group: Arc<dyn Fn() -> Composer>,
+}
+
+impl TransactionCreator {
+    pub fn new(new_group: Arc<dyn Fn() -> Composer>) -> Self {
+        Self { new_group }
+    }
+
+    pub(crate) async fn transaction<F>(
+        &self,
+        composer_method: F,
+    ) -> Result<Transaction, ComposerError>
+    where
+        F: FnOnce(&mut Composer) -> Result<(), ComposerError>,
+    {
+        let mut composer = (self.new_group)();
+        composer_method(&mut composer)?;
+        let built_transactions = composer.build(None).await?;
+
+        built_transactions
+            .last()
+            .map(|tx_with_signer| tx_with_signer.transaction.clone())
+            .ok_or(ComposerError::StateError(
+                "No transactions were built by the composer".to_string(),
+            ))
+    }
+
+    pub async fn payment(&self, params: PaymentParams) -> Result<Transaction, ComposerError> {
+        self.transaction(|composer| composer.add_payment(params))
+            .await
+    }
+
+    pub async fn asset_create(
+        &self,
+        params: AssetCreateParams,
+    ) -> Result<Transaction, ComposerError> {
+        self.transaction(|composer| composer.add_asset_create(params))
+            .await
+    }
+
+    pub async fn asset_transfer(
+        &self,
+        params: AssetTransferParams,
+    ) -> Result<Transaction, ComposerError> {
+        self.transaction(|composer| composer.add_asset_transfer(params))
+            .await
+    }
+
+    pub async fn asset_opt_in(
+        &self,
+        params: AssetOptInParams,
+    ) -> Result<Transaction, ComposerError> {
+        self.transaction(|composer| composer.add_asset_opt_in(params))
+            .await
+    }
+
+    pub async fn asset_opt_out(
+        &self,
+        params: AssetOptOutParams,
+    ) -> Result<Transaction, ComposerError> {
+        self.transaction(|composer| composer.add_asset_opt_out(params))
+            .await
+    }
+
+    // TODO: Rename to `asset_config` to align with Python/TypeScript implementations
+    pub async fn asset_reconfigure(
```

Comment:
I'd rename now.

Reply:
renamed but only in the abstractions im adding in this pr rest need to be broadly aligned during stabilization